packages/key-management/src/types.ts

```diff
@@ -244,4 +277,7 @@ export interface Witnesser {
    * @throws AuthenticationError
    */
   signBlob(derivationPath: AccountKeyDerivationPath, blob: HexBlob, context: SignDataContext): Promise<SignBlobResult>;
+
+  /** Given a transaction with its hash, returns a witnessed transaction. */
+  finalizeTx(args: FinalizeTxArgs): Promise<WitnessedTx>;
```

Comment:
There are a few difference, the witness method only generates the witness set, but the finalize returns a complete transaction, it also executes the externalSigners collection of callbacks, it can also generate dummy witness for transaction size estimation. Now I believe we dont use the witness method other than inside finalizeTx, so I think it could work to just have one method that does everything, I will try this.

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1111/commits/1c10daeb5c43ef636aedebd1aeb221e3c3208588. As a side note, I had issue merging the commit which moved the finalizeTx method to the witnesser and this new commit which merge both methods, this is because I performed a heavy refactor and I had other commit in the way, then it became really hard to separate  (and realized too late I would then have an issue here). The previous commit doesn't make much sense since this commit replaces it, but It would take a lot of work to merge then 😅 . If you feel is unacceptable to keep that commit, I will do the required work to merge them

Reply:
Best to squash any intermediate commits